openshift/Dockerfile.requirements

```diff
@@ -28,40 +26,64 @@ RUN python3 -m pip install pipenv==2023.11.15 \
+    && sed -i '/ansible-core==/s/^/#/g' ./requirements.txt \
     && sed -i '/ansible-runner==/s/^/#/g' ./requirements.txt \
     && sed -i '/ansible-runner-http==/s/^/#/g' ./requirements.txt \
-    && ./pip_find_builddeps.py requirements.txt -o requirements-build.in --append \
-    && pip-compile --output-file=./requirements-build.txt ./requirements-build.in --strip-extras --allow-unsafe \
+    && sed -i '/requests-unixsocket==/s/^/#/g' ./requirements.txt \
     # NOTE: Comment out cryptography and its dependencies from the requirements.txt
-    # and requirements-build.txt files as these packages can't be installed in the
-    # isolated environment of OSBS image build. These packages will be installed
-    # through rpms.
+    # files as these packages can't be installed in the isolated environment of OSBS
+    # image build. These packages will be installed through rpms.
     && sed -i '/cryptography==/s/^/#/g' ./requirements.txt \
     && sed -i '/cffi==/s/^/#/g' ./requirements.txt \
     && sed -i '/pycparser==/s/^/#/g' ./requirements.txt \
+    && sed -i '/maturin==/s/^/#/g' ./requirements.txt \
+    && ./pip_find_builddeps.py requirements.txt -o requirements-build.in --append \
+    # Uncomment ansible-core, ansible-runner, ansible-runner-http and requests-unixsocket,
+    # so that they are re-enabled in the requirements.txt file.
+    && sed -i '/ansible-core==/s/^#//g' ./requirements.txt \
+    && sed -i '/ansible-runner==/s/^#//g' ./requirements.txt \
+    && sed -i '/ansible-runner-http==/s/^#//g' ./requirements.txt \
+    && sed -i '/requests-unixsocket==/s/^#//g' ./requirements.txt \
+    && pip-compile --output-file=./requirements-build.txt ./requirements-build.in --strip-extras --allow-unsafe \
+    # NOTE: Comment out cryptography and its dependencies from the requirements-build.txt
+    # files as these packages can't be installed in the isolated environment of OSBS image
+    # build. These packages will be installed through rpms.
+    && sed -i '/cryptography==/s/^/#/g' ./requirements-build.txt \
     && sed -i '/cffi==/s/^/#/g' ./requirements-build.txt \
     && sed -i '/pycparser==/s/^/#/g' ./requirements-build.txt \
     && sed -i '/maturin==/s/^/#/g' ./requirements-build.txt \
+    # Add ansible-core into a separate requirements-build1-temp.in file to include it
+    # into the requirements-build1.in file.
+    && grep "ansible-core==" ./requirements.in >> ./requirements-build1-temp.in || true \
+    && ./pip_find_builddeps.py requirements-build1-temp.in -o requirements-build1.in --append \
+    && pip-compile --output-file=./requirements-build1.txt ./requirements-build1.in --strip-extras --allow-unsafe \
+    # NOTE: Comment out cryptography and its dependencies from the requirements-build1.txt
+    # files as these packages can't be installed in the isolated environment of OSBS
+    # image build. These packages will be installed through rpms.
+    && sed -i '/cryptography==/s/^/#/g' ./requirements-build1.txt \
```

Comment:
What are the options to alleviate building a new requirements-build.txt?

Reply:
Since there are 3 conflicting versions of `setuptools` needed as part of the build requirements for `ansible-core`, `ansible-runner` and `charset-normalizer`, additional build requirements file is added. Each of the 3 build requirements files are used to install the different versions of `setuptools` in the OSBS environment.

The conflict is explained here: https://github.com/openshift/ansible-operator-plugins/pull/40/files#diff-7ad0e6b82f775968960444bb0baafc660c603428d601e19fdcf45a81623775eeR32-R37